### PR TITLE
Create review frontend

### PR DESCRIPTION
**Pull request by garrickhenne**

### CODE REVIEW

**Comment by garrickhenne** [comment]
NOTE: NPM INSTALL NEEDED AFTER MERGING

File: backend/src/controllers/PropertyController.ts
Comment (garrickhenne) [inline backend/src/controllers/PropertyController.ts:1]: Theres literally no changes here, some dumb things about crlf and lf changed https://stackoverflow.com/questions/1552749/difference-between-cr-lf-lf-and-cr-line-break-types

File: backend/src/services/PropertyService.ts [hunk 1]
```diff
@@ -17,4 +17,8 @@ export const getProperties = (landLordFirstName: string, landLordLastName: strin
 
       return Property.find({ landlordId: { $in: landlordIds } });
     });
+};
+
+export const getProperty = (postalCode: string, streetName: string, streetNum: number): Promise<IProperty | null> => {
+  return Property.findOne({ postalCode, streetName, streetNumber: streetNum }).exec();
```
Comment (garrickhenne) [inline backend/src/services/PropertyService.ts:23]: Allows check to see if property exists already when creating a review

File: backend/src/services/ReviewService.ts [hunk 2]
```diff
@@ -90,23 +90,26 @@ export const createReview = (
...
+        .then(property => {
+          propertyId = property._id;
 
           // ...Finally we have all 2 ids, create Review.
-          return createReviewWithUpdatedValues(title, desc, sentiment, healthSafety, respect, repair, calculatedScore, userId, landlordId)
-            .catch((error) => {
-              console.log(error, "error creating review.");
-              return null;
-            });
+          return createReviewWithUpdatedValues(title, desc, sentiment, healthSafety, respect, repair, calculatedScore, userId, landlordId);
```
Comment (garrickhenne) [inline backend/src/services/ReviewService.ts:106]: Check to see if property already exists, if not create a property before creating review

File: frontend/src/components/LogoutButton.jsx [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
 const LogoutButton = () => {
+  const navigate = useNavigate();
   const { user, logout } = useContext(authContext);
   const handleClick = () => {
     axios.get('/api/logout')
-      .then(() => logout())
+      .then(() => {
+        logout();
+        navigate('/');
+      })
```
Comment (garrickhenne) [inline frontend/src/components/LogoutButton.jsx:14]: fix unrelated to this story

File: frontend/src/components/ReviewForm/InputTextBox.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  };
+
+  return(
+    <div className="text-start gap-y-1 w-auto">
+      <p>{ labelText }</p>
+      <input
+        type="text"
+        placeholder={ placeHolderText }
+        className={`bg-transparent border border-white rounded-sm pl-3 h-9 ${minWidth}`}
+        onChange={ handleChange }
```
Comment (robertshum) [inline frontend/src/components/ReviewForm/InputTextBox.jsx:13]: I'm wondering, will the code become more complicated than setting a value?  If not maybe something like:

```suggestion
onChange={(e)=> setVal(e.target.value)}
```

File: frontend/src/components/ReviewForm/ReviewForm.jsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import axios from 'axios';
+import { useNavigate } from "react-router-dom";
+
+const ReviewForm = () => {
+  const navigate = useNavigate();
+  const { landlord, property, reviewRating } = useContext(newReviewContext);
+
+  const handleSubmit = (e) => {
+    const newUser = {
+      title: reviewRating.title,
```
Comment (robertshum) [inline frontend/src/components/ReviewForm/ReviewForm.jsx:15]: I don't think I did my git revert properly :P I don't know why my form values are not the old ones we agreed on, but at least we're on the same page now!

File: frontend/src/components/ReviewForm/ReviewTextInput.jsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const ReviewTextInput = ({ labelText, placeHolderText, val, setVal }) => {
+  const handleChange = (e) => {
+    setVal(e.target.value);
+  };
+
+  return(
+    <div className="text-start gap-y-1 w-auto">
+      <p>{ labelText }</p>
+      <textarea name="" id="" cols="30" rows="10" placeholder={ placeHolderText } value={val} onChange={handleChange} className="bg-transparent border border-white rounded-sm p-3 w-[90%]"></textarea>
```
Comment (robertshum) [inline frontend/src/components/ReviewForm/ReviewTextInput.jsx:9]: same thing here: https://github.com/garrickhenne/tenant/pull/32/files/1c3dc9dbc905fbb2904276902a14c60e9fdc6202#r1410983687

File: frontend/src/components/ReviewForm/AutoCompleteInput.jsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            setVal(value.postalCode);
+            property.setStreetNumber(value.streetNumber);
+            property.setStreetName(value.streetName);
+          }}
+          isOptionEqualToValue={ (option, value) => option.postalCode === value.postalCode }
+          getOptionLabel={(option) => {
+            return `${option.postalCode} - ${option.streetName} ${option.streetNumber}`;
+          }}
+          options={options}
+          freeSolo
```
Comment (robertshum) [inline frontend/src/components/ReviewForm/AutoCompleteInput.jsx:82]: Great documentary btw

**Review by robertshum — COMMENTED** [review]
Just minor recommendations related to submit handler, but otherwise looks great.

File: frontend/src/components/ReviewForm/InputTextBox.jsx (hunk 4 above)
Comment (garrickhenne) [reply frontend/src/components/ReviewForm/InputTextBox.jsx]: done

File: frontend/src/components/ReviewForm/ReviewTextInput.jsx (hunk 6 above)
Comment (garrickhenne) [reply frontend/src/components/ReviewForm/ReviewTextInput.jsx]: done!

**Pull request merged**